strutil/quantity/quantity.go

```diff
@@ -298,20 +298,28 @@ func formatDurationGeneric(dt float64, min Duration, max Duration, count Places,
 	// Fractional seconds to nanoseconds
 	delta := Duration(dt * float64(Second))
 
-	// If we only render a subset of place values, we need to apply
-	// the render mode to the last rendered place value. We do this
-	// indirectly by moving up the 'min' accuracy.
+	// If we only render a subset of place values (i.e. Compact Mode),
+	// apply the render mode to the last rendered place value. We do
+	// this indirectly by moving up the 'min' accuracy.
 	if count == ShowCompact {
+		// Compact mode has a width limit of 6 runes. In the design of this mode
+		// we allocate 3 runes per place value if the place value only has 2
+		// digits (allowing two places to be rendered). Place values which can
+		// be wider (y, d, ms, us and ns) which means we can only render a single
+		// place for some of the cases.
 		var compactMin Duration
 		for i, key := range keys {
-			// Only if the duration is applicable to this place value
+			// Search for the most significant place value we need to render
 			if delta >= key {
-				if key >= Minute {
-					// Compact mode only renders two places, up until seconds
-					// The indexing is safe as places beyond seconds exist
+				if key >= Minute && key <= Day && delta/key <= 99 {
+					// Compact mode renders two places, if:
+					// 1. Most significant place is d, h or m
+					// 2. If days requires less than 2 digits
 					compactMin = keys[i+1]
 				} else {
-					// Only one place if we are s, ms, us or ns
+					// Compact mode renders single places, if:
+					// 1. Most significant place is y, s, ms, us or ns
+					// 2. If days requires more than 2 digits
```

Comment:
Am I correct in saying that these two conditions are ORed, wheres those in the comment in the other branch are ANDed?

If so, I'd suggest to reformulate a bit this one in order to make it clear:
```suggestion
					// Compact mode renders single places, if either:
					// 1. most significant place is y, s, ms, us or ns
					// 2. days require more than 2 digits
```

Reply:
Both are OR cases. Fixed.